stacks_queues/queue.py

```diff
@@ -23,39 +23,67 @@ def enqueue(self, element):
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == self.buffer_size:
+            raise QueueFullException("Queue is full!")
+
+        if self.size == 0:
+            self.front = 0
+            self.rear = 0
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.empty():
+            raise QueueEmptyException("Queue is empty!")
+    
+        front_ele = self.store[self.front]
+        self.front = (self.front + 1) % self.buffer_size
+        self.size -= 1
+        return front_ele
+
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if not self.store:
+            return None
```

Comment:
The store is always initialized to a non-empty list (full of `Nones`, but non-empty), so this will never be true. Instead, we can check the size or the result of empty().
```suggestion
        if self.empty():
            return None
```